library/src/test/java/com/pengrad/telegrambot/impl/TelegramBotClientTest.java

```diff
@@ -0,0 +1,103 @@
+
+        SendMessage oldRequest = new SendMessage("chatId","ChattieChat").parseMode(ParseMode.MarkdownV2).replyMarkup(oldKeyboard);
+
+        RequestBody oldRequestBody = client.createRequestBody(oldRequest);
+        Buffer oldBuffer = new Buffer();
+        oldRequestBody.writeTo(oldBuffer);
+        String oldBody = oldBuffer.readUtf8();
+
+        String expected = String.join("&",
+                "chat_id=chatId",
+                "text=ChattieChat",
+                "parse_mode=MarkdownV2",
+                "reply_markup="+ URLEncoder.encode("{\"keyboard\":" +
+                        "[" +
+                            "[" +
+                                "{\"text\":\"Button0.1\",\"request_contact\":false,\"request_location\":false}," +
+                                "{\"text\":\"Button0.2\",\"request_contact\":false,\"request_location\":false}" +
+                            "]," +
+                            "[" +
+                                "{\"text\":\"Button1.1\",\"request_contact\":false,\"request_location\":false}," +
+                                "{\"text\":\"Button1.2\",\"request_contact\":false,\"request_location\":false}" +
+                            "]" +
+                        "]," +
+                        "\"resize_keyboard\":true,\"one_time_keyboard\":true,\"selective\":false" +
+                        "}",StandardCharsets.UTF_8.displayName()
+                )
+        );
+
+        Assert.assertEquals(expected, oldBody);
+
+        Keyboard newKeyboard = new ReplyKeyboardMarkup().addRow("Button0.1","Button0.2").addRow("Button1.1","Button1.2").oneTimeKeyboard(true).resizeKeyboard(true);
+        SendMessage newRequest = new SendMessage("chatId","ChattieChat").parseMode(ParseMode.MarkdownV2).replyMarkup(newKeyboard);
+
+        RequestBody newRequestBody = client.createRequestBody(newRequest);
+        Buffer newBuffer = new Buffer();
+        newRequestBody.writeTo(newBuffer);
+        String newBody = newBuffer.readUtf8();
+        Assert.assertEquals(oldBody,newBody);
+    }
+
+    @Test
+    public void sendPollBody() throws IOException {
+        SendPoll request = new SendPoll("chatId", "Huh", "Bird", "Bear", "Chicken");
+        RequestBody requestBody = client.createRequestBody(request);
+        Buffer sink = new Buffer();
+        requestBody.writeTo(sink);
+        Assert.assertEquals("chat_id=chatId&question=Huh&options=%5B%22Bird%22%2C%22Bear%22%2C%22Chicken%22%5D", sink.readUtf8());
+    }
+
+    @Test
+    public void parseSendResponseBadRequest() {
+        String responseFromTelegram = "{\"ok\":false,\"error_code\":400,\"description\":\"Bad Request: message can't be edited\"}";
+        SendResponse response = BotUtils.fromJson(responseFromTelegram,SendResponse.class);
+        Assert.assertNotNull(response);
+        Assert.assertFalse(response.isOk());
+    }
+
+    @Test
+    public void parseNull() {
+        Assert.assertNull(BotUtils.fromJson(null,SendResponse.class));
```

Comment:
Moved the method, though I'm not  fan of static imports due to fact I always think it's a local-method. A test slipped though since a developer implemented a method like in it test-case like:
```java
void assertEquals(Object a, Object b) {
  System.out.println("assertEquals("+a+","+b);
}
```
That's why I have a blocking checkstyle on static imports.